StoreKitUnitTests/TestHelpers/StoreKitTestHelpers.swift

```diff
@@ -57,9 +58,82 @@ extension StoreKitConfigTestCase {
 
 }
 
+extension StoreKitConfigTestCase {
+
+    /// Updates `SKTestSession.storefront` and waits for `Storefront.current` to reflect the change
+    /// This is necessary because the change is aynchronous within `StoreKit`, and otherwise code that depends
+    /// on the change might not see it in time, resulting in race conditions and flaky tests.
+    func changeStorefront(_ new: String) async {
+        testSession.storefront = new
+
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+            // Note: a better approach would be using `XCTestExpectation` and `self.wait(for:timeout:)`
+            // but it doesn't seem to play well with async-await.
+            // Also `toEventually` (Quick nor Nimble) don't support `async`.
+
+            var storefrontUpdateDetected: Bool {
+                get async { await Storefront.current?.countryCode == new }
+            }
+
+            var numberOfChecksLeft = 10
+
+            repeat {
+                if await !storefrontUpdateDetected {
+                    numberOfChecksLeft -= 1
+                    try? await Task.sleep(nanoseconds: 100_000_000) // 0.1 seconds
+                } else {
+                    break
+                }
+            } while numberOfChecksLeft > 0
+
+            let detected = await storefrontUpdateDetected
+            expect(detected).to(beTrue(), description: "Storefront change not detected")
+        }
+    }
+
+}
+
 extension StoreKitConfigTestCase {
 
     static let productID = "com.revenuecat.monthly_4.99.1_week_intro"
     static let lifetimeProductID = "lifetime"
 
 }
+
+extension ProductsFetcherSK1 {
+
+    func product(withIdentifier identifier: String) async throws -> StoreProduct {
+        let products = try await self.products(withIdentifiers: Set([identifier]))
+
+        switch products.count {
+        case 0: throw StoreKitTestError.noProductsFound
+        case 1: return StoreProduct.from(product: products.first!)
+        default: throw StoreKitTestError.multipleProductsFound
+        }
+    }
+
+}
+
+enum StoreKitTestError: Swift.Error {
```

Comment:
Can you put this inside of `StoreKitConfigTestCase` to avoid putting it in the global namespace? It could get confusing with `SKTestError`.

Also this could be `fileprivate`.

Reply:
yep, much better 👍🏼